includes/class-wc-order.php

```diff
@@ -103,25 +108,32 @@ public function payment_complete( $transaction_id = '' ) {
 			if ( ! $this->get_id() ) {
 				return false;
 			}
-			do_action( 'woocommerce_pre_payment_complete', $this->get_id() );
 
-			if ( ! empty( WC()->session ) ) {
-				WC()->session->set( 'order_awaiting_payment', false );
-			}
+			$lock_name = 'order_' . $this->get_id() . '_payment_complete';
+
+			if ( WC_Database_Locks::aquire_lock( $lock_name ) ) {
```

Comment:
@mikejolley sorry I meant the `payment_complete` method that we're talking in right now. If acquiring the lock fails, `payment_complete` returns true.